### PR TITLE
Role Onboarding using Quest Chains

### LINKED ISSUE #1182: Coming soon overlay for quest categories
### What would you like to be added?

Coming soon overlay for quest categories, which are not implemented yet.

### Why is this needed?

Some categories are not implemented yet. But we wanna tease ppl on what's to come.

### PR DESCRIPTION
**Pull request by vidvidvid**
## Overview

**What features/fixes does this PR include?**

Setup Quest Chains for different role paths.

## Todo

- [x] `coming soon` disclaimers on top of `initiation` and `path of the engaged`
- [x] nicer descriptions
- [x]  finish chains design
- [x]  refresh status when uploading proof
- [x]  markdown for quests/quest chain description
- [x]  mint NFT button/functionality

## Improvements

- [ ]  step onto the path button
- [ ]  [overlay with info](https://www.figma.com/proto/h4w7a6N8qtDblQDO4UpfkQ/Seeds-page?page-id=0%3A1&node-id=1220%3A35537&viewport=-3326%2C-1361%2C0.17&scaling=scale-down&starting-point-node-id=1220%3A35537&show-proto-sidebar=1)
- [ ] show completed badges - can be done if we support opensea NFTs on profile nft gallery
- [ ] All progress on `/roles` page

## Closes

Closes #1182

## Assets
Roles Page: 
Screenshot from 2022-10-08 11:05:09

Octo's Path:
Screenshot from 2022-10-08 11:05:45

In Review:
Screenshot from 2022-10-08 09:42:04

To Mint: 
Screenshot from 2022-10-08 11:02:15

Success: 
Screenshot from 2022-10-08 10:53:16

## Preview

https://frontend-pr-1150-mjhnbmqqna-uk.a.run.app/roles

Paired with @dan13ram

### CODE REVIEW

**Review by luxumbra — COMMENTED** [review]
Loving your work folks. ⚔️ 

I won't pass comment on the code but I have made it through the flow as an engaged octo and aside from some style/responsive issues (which shouldn't block this imho), it's frickin awesome. 🥳 

One thing I noticed was that if I hit the left cursor key by accident, i go back but going back to the step i wanted to be on, I lost the text i'd written.
 
On a quick look over on mobile, there's a bunch of responsive improvements that could be made with gutters and font sizes. Not blocking issues but would be nice for our mobile users. :)

Also it would be good to see a back button to allow the user to return to the path selection screen if they happen to change their minds. As it is a flow, it feels odd to have to go to the browsers back button to go back a step in the flow. 
image

Again, not a blocker but this screen could probs be put in a 2 or 3 col grid on mobile with tighter gutters/less whitespace. Would save scrolling for mobile users. 👇🏻 
image


On 1366px and indeed on 1920 these could be a fair bit tighter so they fit within one screen. That extra scroll feels unnecessary. 
image


On desktop, if i mouse over the Quest title, it forces it to two lines when the external link icon appears. 

https://user-images.githubusercontent.com/780350/194718248-92764d81-ebca-4d67-b853-429247b1e0ea.mp4

**Review by luxumbra — APPROVED** [review]
At this point in the day, I'd say LGTM unless @alalonde or @dysbulic have any issues. 🚀

**Review by luxumbra — APPROVED** [review]
Much better. lgtm 🍻 👯

File: packages/web/pages/play/paths/[questchain].tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      </PageContainer>
+    );
+  }
+
+  return (
+    <PageContainer>
+      <HeadComponent
+        title={`MetaGame ${inputQuestChain.name}`}
+        description="MetaGame is a Massive Online Coordination Game! MetaGame has some epic quests going on!"
+        url="https://my.metagame.wtf/quests/path-of-the-engaged"
```
Comment (alalonde) [inline packages/web/pages/play/paths/[questchain].tsx:81]: Looks like this needs to get update?

File: packages/web/lib/hooks/useBoundingRect.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+
+const debounce = (
```
Comment (alalonde) [inline packages/web/lib/hooks/useBoundingRect.ts:3]: Do we not have something like this in utils already? If not this should probably go there

Comment (dan13ram) [reply packages/web/lib/hooks/useBoundingRect.ts:3]: Not sure. I copied this working code from another repo of mine without looking into it 😅

**Pull request merged**